app/utils/epic.server.ts

```diff
@@ -64,21 +64,16 @@ export async function validateUserPurchase({
 		)
 	}
 
-	const noAccessStatuses = ['Refunded', 'Disputed', 'Banned']
+	const validPurchaseStatuses = ['Valid', 'Restricted']
 
-	if (
-		userInfoResult.data.purchases.some(p => noAccessStatuses.includes(p.status))
-	) {
-		throw json(
-			{ status: 'error', error: 'User has a refunded purchase' } as const,
-			{ status: 400 },
-		)
-	}
+	const validPurchases = userInfoResult.data.purchases.filter((p) => 
+		validPurchaseStatuses.includes(p.status)
+	)
```

Comment:
basically we can ignore anything that isn't valid. I don't want to filter them on the server side because there are use cases for knowing the full purchase history.

Reply:
Makes sense!